packages/clerk-js/src/utils/cookies/getCookieDomain.ts

```diff
@@ -0,0 +1,32 @@
+import { createCookieHandler } from '@clerk/shared/cookie';
+
+/**
+ * Determines the eTLD+1 domain, which is where we want the cookies to be set.
+ * This aligns with logic in FAPI, which is important to ensure we don't run into
+ * a scenario where two __client_uat cookies are present.
+ */
+let eTLDPlusOne: string;
+const eTLDCookie = createCookieHandler('__clerk_test_etld');
+export function getCookieDomain(hostname = window.location.hostname) {
+  // only compute it once per session to avoid unnecessary cookie ops
+  if (eTLDPlusOne) {
+    return eTLDPlusOne;
+  }
+
+  const hostnameParts = hostname.split('.');
+
+  // we know for sure that the first entry is definitely a TLD, skip it
+  for (let i = hostnameParts.length - 2; i >= 0; i--) {
```

Comment:
Walking backwards here (from the TLD) so we can construct progressively more specific domains. It should be more efficient going from the full domain to the TLD.

Reply:
How do we solve for localhost?

Reply:
Fixed!

Reply:
@BRKalow would this work for addresses poitning to localhost like `0.0.0.0`?